star-sys/STAR/source/MmapArray.cpp

```diff
@@ -9,51 +9,71 @@ MmapArray::MmapArray() {
     prefix_mmap=NULL;
     file_mmap=NULL;
     file_mmap_length=0;
+    fd=0;
 };
 
-#define PAGE (1<<12)
+#define PAGE ((uint) (1<<12))
 
-uint round_up_to_page(int v) {
-    return v + (PAGE - 1) & !(PAGE - 1);
+uint round_up_to_page(uint v) {
+    return v + (PAGE - 1) & ~(PAGE-1);
 }
 
 int MmapArray::makeMmap(string filename, uint length, uint suffix_padding) {
 
-    int fd = open(filename.c_str(), O_RDONLY);
+    fd = open(filename.c_str(), O_RDONLY);
+    if (fd == -1) {
+        return -1;
+    }
 
     file_mmap_length = round_up_to_page(length + suffix_padding);
-
+    
```

Comment:
Ideally, we'd just `clang-format` the whole project.  Maybe I can come up with a `.clang-format` that makes minimal changes to what's here already, but it's not terribly consistent.
```suggestion

```